### PR TITLE
Support user project override

### PR DESCRIPTION
**Pull request by chyijia**
It's a method that enables users to authenticate with their own OAuth credentials and use the token to send requests to the Build API without enabling the API in their GCP project.

### CODE REVIEW

**Comment by ser-io** [comment]
> For more information, please refer to b/362409289

let's avoid referencing internal bugs in the open source contributions, we can rather add a relevant summary if necessary.

File: frontend/src/host_orchestrator/orchestrator/controller.go
Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/controller.go:40]: this header name is used by GCP, let's use our own name: `X-Cutf-Host-Orchestrator-BuildAPI-Creds-User-Project-ID`.

File: frontend/src/host_orchestrator/orchestrator/controller.go
Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/controller.go:158]: in the HO context, let's use the term `credsUserProjectID`, `projectID` is confusing as it could represent the GCP project the HO is hosted in as well.

File: frontend/src/host_orchestrator/orchestrator/createcvdaction.go
Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:49]: join this two into a new struct:

```
type BuildAPICredentials  struct {
  AccessToken string
  // Add documentation here explaining when this is required.
  UserProjectID string
}
```

**Review by ser-io — CHANGES_REQUESTED** [review]

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:249]: I'm wondering if we can make this simpler (and likewise for the other places where you need to check the start of the query string). 

For example:
```
bool firstQueryAdded = false;
if (!api_key_.empty()) {
  url += ...
  firstQueryAdded = true;
}
if(!project_id_.empty()){
  url += (firstQueryAdded ? "&" : "?") + ...;
  firstQueryAdded = true;
}
```
What do you think?

File: base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp:375]: remove whitespace

File: base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp:377]: remove whitespace

File: frontend/src/host_orchestrator/orchestrator/createcvdaction.go
Comment (denniscy1993) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:159]: Should we set `--project_id` only when `a.buildAPICredentials.AccessToken != ""`?

File: frontend/src/host_orchestrator/orchestrator/createcvdaction.go
Comment (denniscy1993) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:81]: Let's keep this empty line.

File: frontend/src/host_orchestrator/orchestrator/instancemanager.go
Comment (denniscy1993) [inline frontend/src/host_orchestrator/orchestrator/instancemanager.go:269]: Likewise, should we set `--project_id` only when `f.buildAPICredentials.AccessToken != ""`?

**Review by denniscy1993 — CHANGES_REQUESTED** [review]

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (chyijia) [reply base/cvd/cuttlefish/host/libs/web/android_build_api.cpp]: Agreed, thank you for the suggestion!

File: frontend/src/host_orchestrator/orchestrator/createcvdaction.go
Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:41]: nit: `UserProjectID string`, the prefix `Creds` in `CredsUserProjectID` is redundant.

Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:41]: nit: UserProjectID string, the prefix Creds in CredsUserProjectID is redundant.

File: frontend/src/host_orchestrator/orchestrator/createcvdaction.go
Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:40]: nit: Substitute `that enables the Build API` with `that has the Build API enabled`.

Comment (ser-io) [inline frontend/src/host_orchestrator/orchestrator/createcvdaction.go:40]: state what's is the expected value for the field `CredsUserProjectID` when you need to set it, i.e the `GCP project id the AccessToken was created with`.

**Review by ser-io — APPROVED** [review]
Approval from the HO code only. Please wait for approvals from @cjreynol  or @jemoreira on the cvd code.

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:239]: Sorry I just noticed that cvd uses snake_case to name variables. Let's use `first_query_added`.

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:241]: Why did you remove the leading `?`?

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:245]: This line of code is too long, please split it. For example
```
url += ((first_query_added ? ...) +
        std::string("$userProject=") + ...);
```

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:267]: Likewise, why did you remove the leading `?`?

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:271]: Likewise, please split this line of code.

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:362]: Likewise, let's use `first_query_added`.

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/libs/web/android_build_api.cpp:364]: Likewise, why did you remove the leading `?`?

File: base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp
Comment (denniscy1993) [inline base/cvd/cuttlefish/host/commands/cvd/parser/load_configs_parser.cpp:164]: wrong indent, please fix.

**Review by denniscy1993 — CHANGES_REQUESTED** [review]

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (chyijia) [reply base/cvd/cuttlefish/host/libs/web/android_build_api.cpp]: Fixed, thanks for the reminder!

File: base/cvd/cuttlefish/host/libs/web/android_build_api.cpp
Comment (chyijia) [reply base/cvd/cuttlefish/host/libs/web/android_build_api.cpp]: Removed it by accident, thanks for the reminder!

**Review by denniscy1993 — APPROVED** [review]
LGTM. Let's wait for @cjreynol or @jemoreira to review the cvd code.

**Review by cjreynol — APPROVED** [review]

**Comment by denniscy1993** [comment]
@jemoreira, it looks like we are still missing the owner's approval. Could you please review this PR for us?

**Review by jemoreira — APPROVED** [review]

**Pull request merged**
